### DISCUSSION TITLE
Symmetrical Conditions

### DISCUSSION THREAD

**Discussion by YGupta31** [kev-m/pyCoilGen#67] [Q&A]
I am currently working on a project to create a biplanar set of coils for an MRI. I was wondering if there is any way to impose symmetric and anti-symmetric conditions?

**Comment by kev-m** [comment]
I do not fully understand what you are asking. The pyCoilGen application, which is based on the MATLAB [CoilGen](https://github.com/Philipp-MR/CoilGen) project, uses an inversion technique to find the coil winding that matches a provided magnetic field. What symmetric/asymmetric conditions are you after, and in what way can the exiting implementation NOT already create the biplanar coils that you are after? 
@Philipp-MR Can you assist?

**Reply by YGupta31** [reply]
The current biplanar contours are not symmetric on a plane. I have attached an example of one of the coils produced for a z gradient. As the target field I am using is a sphere, I was hoping for a way to only calculate a stream function for on octant of the sphere for a quarter of one of the planes and then apply a symmetric and antisymmetric conditions to the design depending on the direction of the gradient. An example of what I am trying to do can be found in the attached paper.  If there is anything else you would like to know please let me know.
[2639_1_online.pdf](https://github.com/kev-m/pyCoilGen/files/13937752/2639_1_online.pdf)

![plot_half_z_f_2D](https://github.com/kev-m/pyCoilGen/assets/115536745/d54081ae-a472-4594-a0a8-9d7fa6abd744)

**Reply by kev-m** [reply]
Are you referring to the extra loops (in pink and brown) that makes the coil top/bottom symmetric, but not left/right symmetric?

There is no way, at present, to force the stream function. If it is really important to you, you could try and manually manipulate the data in the stream function (see https://github.com/kev-m/pyCoilGen/blob/master/pyCoilGen/helpers/persistence.py#L86 for inspiration) and write out your modified data. You could then use the `sf_source_file` parameter (see the [Load](https://pycoilgen.readthedocs.io/en/stable/configuration.html#load)) parameter documentation.

On the other hand, have you experimented with parameters that affect the [contour generation](https://pycoilgen.readthedocs.io/en/stable/configuration.html#contour-parameters) or [stream function generation](https://pycoilgen.readthedocs.io/en/stable/configuration.html#stream-function), etc?

Is the current error in calculated versus requested field too large for you? What is your up-front expectation?

**Comment by Philipp-MR** [comment]
Hi, for the unsymmetry, check if your target field and current carrying surface are symmetric. The solution should be symmetric, even without symmetry constraints.

**Reply by Philipp-MR** [reply]
Also, be seeing your image, it might be the case that the stream function is symmetric, but some loops are deleted. Check the loop deletion criterias..
